packages/story-editor/src/components/canvas/editLayer.js

```diff
@@ -156,9 +161,9 @@ function EditLayerForElement({ element, showOverflow }) {
           ref={setRef}
         />
       )}
-      {EditMenu && (
```

Comment:
The idea of this was that `EditMenu` can be any `EditMenu` and not only trimming edit menu, even though currently we have only trimming.
Is there any reason to change this?

cc @barklund as you originally created this architecture.

Reply:
> Is there any reason to change this?

The reason is that the `VideoEdit` component currently has a dependency on `isTrimMode`/ `useVideoTrim`

With the elements being moved to `element-library` it cannot use that hook from the editor.

This change moves this out of `VideoEdit` to this component here.

Any suggestions to make this more generic are welcome.

Reply:
Good question! This is a tricky one.

`EditMenu` is eventually returning only `<VideoTrimmer />; `  if `isTrimMode` true, and its only being used for video.

https://github.com/GoogleForCreators/web-stories-wp/blob/8f1a9d71d66d85dc8f02a000847d78272d115966/packages/story-editor/src/elements/video/editMenu.js#L28-L32

Therefore I just used that condition here instead of passing the component as prop.

> The idea of this was that EditMenu can be any EditMenu and not only trimming edit menu

If we want to use `EditMenu` component, we may have to do something like this.

```js
  const menu = (
    <EditMenu
        videoTrimmer={ <VideoTrimmer /> }
        isTrimMode={ isTrimMode }
    />
  )

  return menu && (
    <FooterArea showOverflow>
      { menu }
    </FooterArea>
  )
```

because `<VideoTrimmer>` and `useVideoTrim` live in `story-editor`.

Currently, it's just video, but as you said, let's say that in the future we want to add more options for the `EditMenu` for something else, we again will have to update this part of the code anyways.

Thoughts?

Reply:
Hm, I suppose the question is also -- is the Video Trimmer editor-specific or element specific? If it's element-specific and should be supported by default for the video element, then the trimmer itself should also live in the `elements` package perhaps. Thoughts?

Reply:
To me, it looks element-specific, but since its using a lot of hooks and components from `story-editor`, it wouldn't be easy to move it to the element package.

https://github.com/GoogleForCreators/web-stories-wp/blob/main/packages/story-editor/src/components/videoTrim/videoTrimmer.js

Reply:
I also feel like ideally it would be element-specific, however, this would need a lot of reworking indeed.
And allowing different `EditMenu` options also makes sense only if those `EditMenu` items actually live inside the elements' library, if the `EditMenu` content (e.g. video trimmer) lives in the editor for now anyway, then I think what you have currently makes sense.

Might be good to create a potential follow-up issue for future to look into options of moving the video trimmer into the elements package, just to have it there somewhere.

Thoughts?

Reply:
Yes, creating a follow-up issue https://github.com/GoogleForCreators/web-stories-wp/issues/10867 . Let's deal with it separately.